cosmoz-grouped-list.js

```diff
@@ -99,20 +100,93 @@
 		 */
 		_renderedItems: null,
 
+		/**
+		 * Instances of Template that are NOT available for reuse at the moment.
+		 */
+		_usedInstances: [],
+
+		/**
+		 *  Instances of Template that are available for reuse at the moment.
+		 */
+		_reusableInstances: [],
+
+		/**
+		 * Polymer `created` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		created() {
 			this._templateSelectors = [];
 			this._renderedItems = [];
-			this._physicalItens = [];
 		},
 
+		/**
+		 * Polymer `attached` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		attached() {
-			this._isAttached = true;
+			this._templatesObserver = Polymer.dom(this.$.templates)
+				.observeNodes(this._onTemplatesChange.bind(this));
 			this._debounceRender();
 		},
 
+		/**
+		 * Polymer `detached` livecycle function.
+		 *
+		 * @return {void}
+		 */
 		detached() {
-			this._isAttached = false;
+			if (this._templatesObserver) {
+				Polymer.dom(this).unobserveNodes(this._templatesObserver);
+				this._templatesObserver = null;
+			}
 			this.cancelDebouncer('render');
+
+
+			this._resetAllTemplates();
+
+			this._usedInstances = [];
+			this._reusableInstances = [];
+
+		},
+		_onTemplatesChange(change) {
+			if (!Array.isArray(this._ctors) || this._ctors.length === 0) {
+				const templates = Array.from(change.addedNodes)
+					.filter(n => n.nodeType === Node.ELEMENT_NODE && n.tagName === 'TEMPLATE' && n.matches('[type]'));
```

Comment:
I agree that using `id="itemTemplate"` or `id="groupTemplate"` was not a good idea to identify the templates, since the `id` attribute should be unique within a document.
But `type` is not standard for `template` element so it can be confusing.
Since we already use `class` attribute in `cosmoz-omnitable` for `class="cell"` and `class="header"` templates, we should also use the `class` attribute for item and group templates.

Reply:
@plequang @megheaiulian I have updated the code, tests, demo, omnitable PR. to select with class instead of type.